src/molecule_plugins/vagrant/modules/vagrant.py

```diff
@@ -259,9 +259,12 @@
       {% if instance.provider == "vsphere" %}
       {{ instance.provider | lower }}.memory_mb = {{ instance.memory }}
       {{ instance.provider | lower }}.cpu_count = {{ instance.cpus }}
-      {% elif instance.provider.startswith('vmware_') %}
+      {% elif instance.provider.startswith('vmware_') and instance.provider != 'vmware_esxi' %}
       {{ instance.provider | lower }}.vmx['memsize'] = {{ instance.memory }}
       {{ instance.provider | lower }}.vmx['numvcpus'] = {{ instance.cpus }}
+      {% elif instance.provider == 'vmware_esxi' %}
```

Comment:
Why not putting this check before the ``vmware_`` test ? I may need more coffee but I guess this should avoid adding ``and instance.provider != 'vmware_esxi``

Reply:
Shame on me. Will try.

Reply:
Thanks for testing. I'm going to merge the PR then.